File: 1582-special-positions-in-a-binary-matrix/1582-special-positions-in-a-binary-matrix.java
class Solution {
    public int numSpecial(int[][] mat) {
        int m = mat.length;
        int n = mat[0].length;
        int ans = 0;

        int[] ones = new int[n];
        List<Integer> candidates = new ArrayList<>();

        for (int i = 0; i < m; i++) {
            List<Integer> temp = new ArrayList<>();
            for (int j = 0; j < n; j++) {
                if (mat[i][j] == 1) {
                    temp.add(j);
                    ones[j]++;
                }
            }
            if (temp.size() == 1) {
                candidates.add(temp.get(0));
            }
        }

        for (int col : candidates) {
            if (ones[col] == 1) {
                ans++;
            }
        }

        return ans;
    }
}

File: 1582-special-positions-in-a-binary-matrix/1582-special-positions-in-a-binary-matrix.cpp
class Solution {
public:
    int numSpecial(vector<vector<int>>& mat) {
        int m = mat.size();
        int n = mat[0].size();
        int ans = 0;
        vector<int> ones(n,0);
        vector<int> candidates;
        for(int i = 0; i < m; ++i) {
            vector<int> temp = {};
            for(int j=0; j < n; ++j) {
                if(mat[i][j] == 1) {
                    temp.push_back(j);
                    ones[j] += 1;
                }
            }
            if(temp.size() == 1) candidates.push_back(temp[0]);
        }
        for(auto&v:candidates) {
            if(ones[v] == 1) ans++;
        }
        return ans;
        
    }
};